x/outline-connectivity-app/shared_backend/index.ts

```diff
@@ -0,0 +1,25 @@
+import { ConnectivityTestInput, ConnectivityTestResult } from "./generated/types"
+
+export function from(backend: Invokable) {
+  return {
+    connectivityTest: invocationFactory<ConnectivityTestInput, ConnectivityTestResult[]>(backend, "ConnectivityTest")
```

Comment:
It seems like someone would have to modify this code in order to add a new method.

It's not clear what are all the places where someone would need to touch to make a method a available, but I was hoping it would be a single place.

We are also introducing a significant amount of complexity by coupling the IPC logic with the specific service we are providing. It makes the logic harder to understand, extend and distracts from highlighting the SDK.

For an IPC system, I was hoping to see first a primitive just to pass messages around, like we see in Flutter, Electron and Cordova.

We should be able to write Typescript frontend code that reads like this:
```ts
  channel = new MethodChannel('my.namespace');  
  ...
  result, error = await channel.invoke('testConnectivity', {config: '...'});
```

That's a building block that is decoupled from the specific backend service we are providing and would make it easier to understand the IPC system.

The server side is trivial, as it's just dispatching based on the method name. To add a new method, just update the dispatcher.

Once we have that, we can then build upon it and add type safety. I can see a dispatcher that dispatches to different methods using reflection given an interface, and code to make the invoke calls can be generated from the same interface. That's pretty much what Google's gRPC does. There are established patterns for this.

Can you refactor the code so there's a clear IPC mechanism that is independent of the backend service? It's fine if we end up needing to drop type safety for now.

/cc @jyyi1 since we discussed this at length before and has thoughts too.